### ISSUE TITLE
flutter_test is missing the solo feaure

### ISSUE THREAD

**Issue by alextekartik** [flutter/flutter#35768]
Flutter 1.5.4-1.8.1

Although mentionned in the comment, the `solo` flag is not a parameter of the`test` and `group` function. Here for test:
https://github.com/flutter/flutter/blob/c8f37a2f49a4637b3f896aef4749a343d06d09fb/packages/flutter_test/lib/src/test_compat.dart#L154-L179

**Comment by ketanm** [comment]
@jonahwilliams may know?

**Comment by natebosch** [comment]
The docs for `group` mention `solo`, `timeout`, `testOn`, `tags`, and `onPlatform` which all are missing.

https://github.com/flutter/flutter/blob/36c37ccab5e089e9add6e38ccbcf1c1fffa13ef5/packages/flutter_test/lib/src/test_compat.dart#L235

https://dart-lang.github.io/linter/lints/comment_references.html would help catch these.

**Comment by alextekartik** [comment]
I created a PR https://github.com/flutter/flutter/pull/36604 for this simple change. I have not added the other missing arguments for group (solo is my only need). Thanks!

**Comment by jonahwilliams** [comment]
Discussed on the PR:

I don't think we should add this feature. Individual tests can be run from IDEs and command line, or by commenting out other tests. solo doesn't seem to add a lot of value over this.

**Comment by natebosch** [comment]
@jonahwilliams - IMO we should keep this issue open until we remove the confusing documentation describing parameters which don't exist.

**Comment by jonahwilliams** [comment]
Yes, you're right - I'll open a PR to drop the descriptions there

**Comment by alextekartik** [comment]
In complex unit tests where test files include other test files in different context (so the same test is ran in différent contexts), solo is sometimes the only convenient debugging solution (you can control which file is ran from the IDE but not which inner test from another file is ran). Commenting other tests is definitely not a viable solution.

The solo feature was added on purpose for this (it was deprecated since the beginning, too bad we don't have a better meta keyword for this). Personnally I use this feature a lot for my database solutions (sqflite, sembast). Fortunately when using test_driver, i still have the solo feature (and there is no alternative here to run a single test) but for regular flutter test, I always have to run all the tests when I want to debug a single one (as i said the test is in another file than the file started).

solo is part of the test_api test (https://pub.dev/documentation/test_api/0.1.0/test_api/test.html) function, at some point I guess they will share the same API) and I don't understand why flutter (and only flutter as solo is fine in a VM and browser context) is removing it explicitely.

**Comment by eseidelGoogle** [comment]
For better or worse, Flutter had to write a custom test embedder to integrate with package:test back in the day.  I'm sure our package:test integration is incomplete (lacking features like this one).

My suspicion is we're unlikely to get to this bug anytime soon, but others should feel free to improve Flutter's test integration:
https://github.com/flutter/flutter/tree/master/packages/flutter_test/lib/src

**Comment by markusaksli-nc** [comment]
Seems like the comments and documentation for `solo` have been removed as discussed, leaving this open as a new feature request. Feel free to close if there is no intention to re-implement this.
